config/src/main/java/net/spals/appbuilder/config/provider/TypesafeConfigurationProvider.java

```diff
@@ -97,7 +108,28 @@ public Double get() {
             @Override
             public String get() {
                 try {
-                    return config.getString(key.getRawKey());
+                    final String stringProperty = config.getString(key.getRawKey());
+                    // Check to see if the string property is encrypted
+                    final Matcher encryptedMatcher = ENCRYPTED_PATTERN.matcher(stringProperty);
+                    if (encryptedMatcher.matches()) {
+                        final String encryptedStringProperty = encryptedMatcher.group(1);
+                        try {
+                            // KeyStores may be expensive to set up, so ensure that we only load it once
+                            // (or, at least, try our best to ensure)
+                            final KeyStore keyStore = Optional.ofNullable(keyStoreRef.get())
```

Comment:
I believe you can replace this with 
Suppliers.memoize(() -> KeyStoreProvider.createKeyStore(config)) which should simplfy some of this code